### PR TITLE
Rubicon: add native support

### PR DESCRIPTION
**Pull request by SerhiiNahornyi**

### CODE REVIEW

File: src/main/java/org/prebid/server/bidder/rubicon/RubiconBidder.java [hunk 1]
```diff
@@ -276,13 +279,13 @@ private List<Imp> extractValidImps(BidRequest bidRequest, List<BidderError> erro
     }
 
     private static boolean isValidType(Imp imp) {
-        return imp.getVideo() != null || imp.getBanner() != null;
+        return imp.getVideo() != null || imp.getBanner() != null || imp.getXNative() != null;
```
Comment (And1sS) [inline src/main/java/org/prebid/server/bidder/rubicon/RubiconBidder.java:282]: Maybe `ObjectUtils.anyNotNull(imp.getVideo(), imp.getBanner(), imp.getXNative());`?

File: src/main/java/org/prebid/server/bidder/rubicon/RubiconBidder.java [hunk 2]
```diff
@@ -1036,6 +1045,61 @@ private static int resolveNotStandardSizeForInstl(Format size) {
...
     }
 
+    private Native makeNative(Imp imp, List<BidderError> errors) {
+        final Native xNative = imp.getXNative();
+        final String version = ObjectUtil.getIfNotNull(xNative, Native::getVer);
+        if (StringUtils.equalsAny(version, "1.0", "1.1")) {
+            return xNative;
+        }
+        final String nativeRequest = xNative.getRequest();
+        final JsonNode requestAsNode = nodeFromString(nativeRequest);
```
Comment (And1sS) [inline src/main/java/org/prebid/server/bidder/rubicon/RubiconBidder.java:1055]: final JsonNode `requestNode`

File: src/main/java/org/prebid/server/bidder/rubicon/RubiconBidder.java [hunk 3]
```diff
@@ -1036,6 +1045,61 @@ private static int resolveNotStandardSizeForInstl(Format size) {
...
+
+    public final JsonNode nodeFromString(String stringValue) {
+        try {
+            return StringUtils.isNotBlank(stringValue) ? mapper.mapper().readTree(stringValue) : null;
+        } catch (Exception e) {
+            return null;
+        }
+    }
+
+    private void validateNativeRequest(JsonNode requestAsNode) {
```
Comment (And1sS) [inline src/main/java/org/prebid/server/bidder/rubicon/RubiconBidder.java:1082]: `requestNode`

File: src/test/java/org/prebid/server/bidder/rubicon/RubiconBidderTest.java [hunk 4]
```diff
@@ -216,11 +217,11 @@ public void makeHttpRequestsShouldFilterAllImpressionsAndReturnErrorMeessagesWit
 
         // then
         assertThat(result.getErrors()).hasSize(3)
```
Comment (And1sS) [inline src/test/java/org/prebid/server/bidder/rubicon/RubiconBidderTest.java:219]: Please, remove call to `hasSize()` and replace `containsExactly` with `containsExactlyInAnyOrder`

**Review by And1sS — CHANGES_REQUESTED** [review]

File: src/main/java/org/prebid/server/bidder/rubicon/RubiconBidder.java (hunk 1 above)
Comment (SerhiiNahornyi) [reply src/main/java/org/prebid/server/bidder/rubicon/RubiconBidder.java:282]: Nice catch!

**Review by And1sS — APPROVED** [review]

**Pull request merged**